GenderPayGap.WebUI/Controllers/SubmitController.CheckData.cs

```diff
@@ -174,11 +175,20 @@ public async Task<IActionResult> CheckData(ReturnViewModel postedReturnViewModel
             }
 
             Return postedReturn = submissionService.CreateDraftSubmissionFromViewModel(postedReturnViewModel);
+            Return databaseReturn = submissionService.GetReturnFromDatabase(postedReturnViewModel.OrganisationId, postedReturnViewModel.AccountingDate.Year);
 
             SubmissionChangeSummary changeSummary = null;
-            Return databaseReturn = submissionService.GetSubmissionById(postedReturnViewModel.ReturnId);
             if (databaseReturn != null)
             {
+                if (databaseReturn.ReturnId != postedReturnViewModel.ReturnId)

```

Comment:
What does this check for?

Reply:
this checks if the report has already been submitted (not this report - different id). Basically you can edit an old report (it will be marked as retired later) but can't have 2 submitted  reports at the same time.